src/sinks/util/datagram.rs

```diff
@@ -32,53 +35,87 @@ pub async fn send_datagrams<E: Encoder<Event, Error = vector_lib::codecs::encodi
     mut socket: DatagramSocket,
     transformer: &Transformer,
     encoder: &mut E,
+    chunker: &Option<Chunker>,
     bytes_sent: &<BytesSent as RegisterInternalEvent>::Handle,
 ) {
     while let Some(mut event) = input.next().await {
-        let byte_size = event.estimated_json_encoded_size_of();
-
         transformer.transform(&mut event);
-
         let finalizers = event.take_finalizers();
         let mut bytes = BytesMut::new();
 
         // Errors are handled by `Encoder`.
         if encoder.encode(event, &mut bytes).is_err() {
+            finalizers.update_status(EventStatus::Errored);
```

Comment:
This seems to have been missing; the Encoder does emit logs on error but does not update the status of the `EventFinalizers`.

Let me know if this was an intended behavior.